ifrs17/Test/QueriesTest.ipynb

```diff
@@ -1121,10 +1121,205 @@
+                "\n                                                                Func<T, string> identityExpression)",
+                "\nwhere T : IWithYearMonthAndScenario",
+                "\n{",
+                "\n    Activity.Start();",
+                "\n    await Workspace.UpdateAsync(testData);",
+                "\n    var firstParameter = (await Workspace.LoadCurrentParameterAsync<T>(args, identityExpression)).First();",
+                "\n    ",
+                "\n    //Check Current Period",
+                "\n    firstParameter.Value.Year.Should().Be(expectedCurrentPeriod.Year);",
+                "\n    firstParameter.Value.Month.Should().Be(expectedCurrentPeriod.Month);",
+                "\n    firstParameter.Value.Scenario.Should().Be(expectedCurrentPeriod.Scenario);",
+                "\n    ",
+                "\n    await Workspace.DeleteAsync(Workspace.Query<T>().ToArray());",
+                "\n    return Activity.Finish();",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Partner Rating"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var args = new Args(\"CH\",2020,9,Periodicity.Monthly,null);",
+                "\nvar testData = new PartnerRating{ Partner = \"PT1\", Year = 2020, Month = 9};",
+                "\nvar activity = await CheckLoadCurrentParameterAsync<PartnerRating>(args, testData.RepeatOnce(), (2020, 9, null), x => x.Partner);",
+                "\nactivity"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var args = new Args(\"CH\",2020,9,Periodicity.Monthly,null);",
+                "\nvar testData = new PartnerRating{ Partner = \"PT1\", Year = 2020, Month = 9};",
+                "\nawait Workspace.UpdateAsync(testData);",
+                "\n(await Workspace.LoadCurrentParameterAsync<PartnerRating>(args, x => x.Partner)).First()"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "activity.Status.Should().Be(ActivityLogStatus.Succeeded);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var args = new Args(\"CH\",2020,9,Periodicity.Monthly,scenario);",
+                "\nvar testData = new PartnerRating{ Partner = \"PT1\", Year = 2020, Month = 9, Scenario = scenario };",
```

Comment:
you should have more than one item in testData, so you can check which one is picked by the query.